File: Ruben - Modulo 2/prAltura/src/MainPaises.java
import alturas.*;

import java.io.FileNotFoundException;

public class MainPaises {
    public static void main(String[] args) {
        Paises paises = new Paises();
        Pais[] paisesFiltrado;

        try {
            paises.leePaises("prAltura/src/alturas.txt");
        } catch (FileNotFoundException e) {
            System.out.println("No existe el fichero.");
        }

        System.out.println("Mayores que 1.77");
        for (Pais p: paises.selecciona(new MayoresQue(1.77))) {
            System.out.println(p);
        }

        System.out.println("Menores que 1.77");
        for (Pais p: paises.selecciona(new MenoresQue(1.77))) {
            System.out.println(p);
        }

        System.out.println("De Europa");
        for (Pais p: paises.selecciona(p -> p.getContinente().equals("Europe"))) {
            System.out.println(p);
        }
    }
}

File: Ruben - Modulo 2/prAlturaKT/src/MainPaises.kt
import alturas.MayoresQue
import alturas.MenoresQue
import alturas.Pais
import alturas.Paises

import java.io.FileNotFoundException

object MainPaises {
    @JvmStatic
    fun main(args: Array<String>) {
        val paises = Paises()
        val paisesFiltrado: Array<Pais>

        try {
            paises.leePaises("prAltura/src/alturas.txt")
            paises.paises.filter { pais -> pais.nombre.startsWith("E") }
        } catch (e: FileNotFoundException) {
            println("No existe el fichero.")
        }

        println("Mayores que 1.77")
        for (p in paises.selecciona(MayoresQue(1.77))) {
            println(p)
        }

        println("Menores que 1.77")
        for (p in paises.selecciona(MenoresQue(1.77))) {
            println(p)
        }

        /*
        println("De Europa")
        for (p in paises.selecciona { p -> p.continente == "Europe" }) {
            println(p)
        }
        */
    }
}
